config.m4

```diff
@@ -484,36 +525,50 @@ if test "$PHP_MONGODB" != "no"; then
 
   if test "$PHP_LIBMONGOC" = "no"; then
     backup_srcdir=${srcdir}
-    srcdir=${srcdir}/src/libmongoc/
-    m4_include(src/libmongoc/build/autotools/Versions.m4)
-    srcdir=${backup_srcdir}
+    if test -d "${srcdir}/src/libmongoc/"; then
+      prefixdir=${srcdir}
+    else
+      prefixdir=${srcdir}/ext/mongodb
+    fi
+    srcdir=${prefixdir}/src/libmongoc
+    sinclude(src/libmongoc/build/autotools/Versions.m4)
+    sinclude(ext/mongodb/src/libmongoc/build/autotools/Versions.m4)
     MONGOC_API_VERSION=1.0
     AC_SUBST(MONGOC_MAJOR_VERSION)
     AC_SUBST(MONGOC_MINOR_VERSION)
     AC_SUBST(MONGOC_MICRO_VERSION)
     AC_SUBST(MONGOC_API_VERSION)
     AC_SUBST(MONGOC_VERSION)
-    AC_OUTPUT($srcdir/src/libmongoc/src/mongoc/mongoc-config.h)
-    AC_OUTPUT($srcdir/src/libmongoc/src/mongoc/mongoc-version.h)
+    AC_OUTPUT($prefixdir/src/libmongoc/src/mongoc/mongoc-config.h)
+    AC_OUTPUT($prefixdir/src/libmongoc/src/mongoc/mongoc-version.h)
     if test "x$with_zlib" = "xbundled"; then
-      AC_OUTPUT($srcdir/src/libmongoc/src/zlib-1.2.11/zconf.h)
+      AC_OUTPUT($prefixdir/src/libmongoc/src/zlib-1.2.11/zconf.h)
     fi
+    srcdir=${backup_srcdir}
   fi
   if test "$PHP_LIBBSON" = "no"; then
     backup_srcdir=${srcdir}
-    srcdir=${srcdir}/src/libbson/
-    m4_include(src/libbson/build/autotools/Versions.m4)
-    srcdir=${backup_srcdir}
+    if test -d "${srcdir}/src/libbson/"; then
+      prefixdir=${srcdir}
+    else
+      prefixdir=${srcdir}/ext/mongodb
+    fi
+    srcdir=${prefixdir}/src/libbson
+    sinclude(src/libbson/build/autotools/Versions.m4)
+    sinclude(ext/mongodb/src/libbson/build/autotools/Versions.m4)
     BSON_API_VERSION=1.0
     AC_SUBST(BSON_MAJOR_VERSION)
     AC_SUBST(BSON_MINOR_VERSION)
     AC_SUBST(BSON_MICRO_VERSION)
     AC_SUBST(BSON_API_VERSION)
     AC_SUBST(BSON_VERSION)
-    AC_OUTPUT($srcdir/src/libbson/src/bson/bson-config.h)
-    AC_OUTPUT($srcdir/src/libbson/src/bson/bson-version.h)
+    AC_OUTPUT($prefixdir/src/libbson/src/bson/bson-config.h)
+    AC_OUTPUT($prefixdir/src/libbson/src/bson/bson-version.h)
+    srcdir=${backup_srcdir}
   fi
 
+  EXTRA_LIBS="$EXTRA_LIBS $MONGODB_SHARED_LIBADD"
```

Comment:
Without this, extra libs like snappy or zlib aren't passed on to PHP.

Reply:
Is this ignored for shared builds?

I see bits of `EXTRA_LIBS` in PHP core but it's oddly absent from any extension `config.m4` files. Given that many of those link extra libraries and can also be built-in, is there a reason for that?

Reply:
We discussed why `EXTRA_LIBS` is needed. Long-term, let's see if we can use `PHP_ADD_LIBPATH()`.